pkg/auth/token_handler.go

```diff
@@ -0,0 +1,259 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"go.uber.org/zap"
+	authv1 "k8s.io/api/authentication/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"knative.dev/eventing/pkg/apis/feature"
+	"knative.dev/pkg/configmap"
+	configmapinformer "knative.dev/pkg/configmap/informer"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/system"
+)
+
+type OIDCTokenHandler struct {
+	logger               *zap.Logger
+	kubeClient           kubernetes.Interface
+	provider             *oidc.Provider
+	providerSyncInterval time.Duration
+	providerLock         sync.RWMutex
+	configMapWatcher     configmap.Watcher
+	featureStore         *feature.Store
+}
+
+func NewOIDCTokenHandler(ctx context.Context, kubeClient kubernetes.Interface, logger *zap.Logger) *OIDCTokenHandler {
+	// setup config map watcher, to get informed on changes on the features
+	// configmap to resync the provider config
+	configMapWatcher := configmapinformer.NewInformedWatcher(kubeClient, system.Namespace())
```

Comment:
I'd take this as parameter, some components might have this already instantiated and we don't need another one

Reply:
removed the watcher, as not needed anymore